### PR TITLE
setup dev enviroment in mac os 

### PR DESCRIPTION
**Pull request by CodingCat**
## What changes were proposed in this pull request?

This PR adds Dockerfile and docs about how to build dev environment in mac os 

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: dev/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+RUN apt install -y ccache ninja-build checkinstall libssl-dev libboost-all-dev libdouble-conversion-dev libgoogle-glog-dev libbz2-dev libgflags-dev libgmock-dev libevent-dev liblz4-dev libzstd-dev libre2-dev libsnappy-dev  liblzo2-dev bison flex wget
+ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+ENV PATH=$JAVA_HOME/bin:$PATH
+
+RUN wget https://archive.apache.org/dist/spark/spark-3.2.2/spark-3.2.2-bin-hadoop3.2.tgz
+RUN tar -zxvf spark-3.2.2-bin-hadoop3.2.tgz
+ENV PATH=/spark-3.2.2-bin-hadoop3.2/bin:$PATH
+
+
+RUN ln -fs /usr/share/zoneinfo/America/Los_Angeles /etc/localtime
```
Comment (PHILO-HE) [inline dev/Dockerfile:15]: It's user-location-specific command. Can it be removed or replaced by other command to adapt to different users?

**Comment by Yohahaha** [comment]
have you try to build on macOS directly without Docker?

**Comment by CodingCat** [comment]
yes, tons of issues

**Comment by FelixYBW** [comment]
Let's close this since it doesn't work

**Pull request closed without merging**